### ISSUE TITLE
No output from wp when run from a wordpress install with problems

### ISSUE THREAD

**Issue by technicalpickles** [wp-cli/wp-cli#2681]
I'm trying to run `wp` in a wordpress install that is having problems (ie everything throws 500), and most incantations are generating no output. Only `wp --info` seems to do anything:

```
$ wp --info
PHP binary:	/usr/bin/php5
PHP version:	5.5.9-1ubuntu4.14
php.ini used:	/etc/php5/cli/php.ini
WP-CLI root dir:	phar://wp-cli.phar
WP-CLI packages dir:	/home/technicalpickles/.wp-cli/packages/
WP-CLI global config:	
WP-CLI project config:	
WP-CLI version:	0.23.0
$ wp
$
```

If I'm any other wordpress install on this server, or anywhere else, it works as expected. I understand the wordpress install is broken, so it might be expected that wp wouldn't work, but I would expect an error message.

I think I was on 0.18, and figured I'd update to latest (0.23.0), but I had the same problem. 

```
# What is your system?
$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 14.04.2 LTS
Release:	14.04
Codename:	trusty

$ uname -a
Linux <redacted> 3.13.0-43-generic #72-Ubuntu SMP Mon Dec 8 19:35:06 UTC 2014 x86_64 x86_64 x86_64 GNU/Linux

# Which version of PHP are you using?
$  which -a php
/usr/bin/php
$ php -v
PHP 5.5.9-1ubuntu4.14 (cli) (built: Oct 28 2015 01:34:46) 
Copyright (c) 1997-2014 The PHP Group
Zend Engine v2.5.0, Copyright (c) 1998-2014 Zend Technologies
    with Zend OPcache v7.0.3, Copyright (c) 1999-2014, by Zend Technologies

# Are you running suhosin? If so, make sure you've added `suhosin.executor.include.whitelist = phar` to your php.ini
$ php -m | grep -i suhosin


# Which version of WordPress are you using?
$ grep '^\$wp_version\s' wp-includes/version.php
$wp_version = '4.5';

# Can you share the results of which wp ?
$ which -a wp
/usr/local/bin/wp
$ stat $(which wp)
  File: ‘/usr/local/bin/wp’
  Size: 4039990   	Blocks: 7896       IO Block: 4096   regular file
Device: fd01h/64769d	Inode: 1187863     Links: 1
Access: (0775/-rwxrwxr-x)  Uid: (    0/    root)   Gid: (    0/    root)
Access: 2016-04-22 19:22:35.294132000 -0400
Modify: 2016-04-22 19:21:32.362132000 -0400
Change: 2016-04-22 19:22:21.702132000 -0400
 Birth: -

# Are you running any packages? If one is causing a problem, you can use `--skip-packages` to skip loading them
$ wp package list
+------+-------------+---------+---------+
| name | description | authors | version |
+------+-------------+---------+---------+
+------+-------------+---------+---------+
```

There are some prior issues around this like https://github.com/wp-cli/wp-cli/issues/78 and https://github.com/wp-cli/wp-cli/issues/72 but they end up being install issues.

**Comment by technicalpickles** [comment]
I tracked down my specific wordpress problem to an automatic update of jetpack breaking everything (oh the irony). Once I disabled it (ie renaming it), the site started working, and so did `wp`.

If there's any tests I can do to help debug with the bad plugin in place, let me know.

**Comment by technicalpickles** [comment]
> I tracked down my specific wordpress problem to an automatic update of jetpack breaking everything (oh the irony). Once I disabled it (ie renaming it), the site started working, and so did wp.

Looks like it was this, for anyone wondering: https://wordpress.org/support/topic/fatal-error-when-updating-to-jetpack-40-read-this

**Comment by danielbachhuber** [comment]
Glad you figured it out 👍

**Comment by technicalpickles** [comment]
Would it be worth keeping this open? There were a couple prior of previous issues where wp doesn't output anything if there's a problem with the install. I think if it could output the problem it ran into, that'd be super useful for debugging.

**Comment by danielbachhuber** [comment]
> Would it be worth keeping this open? There were a couple prior of previous issues where wp doesn't output anything if there's a problem with the install.

Given the problem is ultimately a fatal error, it's not a problem we can fix in WP-CLI.

> I think if it could output the problem it ran into, that'd be super useful for debugging.

Yep, see cb91089ee2fd62e2e87accf50221fbebad04671d (added to http://wp-cli.org/docs/bug-reports/)

**Comment by ymazuz** [comment]
So, I can see why you closed this, but I’m having a similar issue with 0.24.1. As I see it, even in the scenario above, the problem isn’t WP-CLI exiting with a fatal error. It’s the fact that it’s doing so silently.

I discovered my problem when I made a change to a local site, typed `wp shell`, and nothing happened. I thought WP-CLI was broken, downloaded and reinstalled it, got the same error, and then finally opened a web browser and saw error messages. Fixed my bugs and `wp shell` worked. 

Great! So the fatal error was my site’s fault. But a little later I was in a WP-CLI interactive shell and mistyped a function name, and it just exited. The exiting is expected, but it used to spit out ‘Fatal error: call to undefined function…” It doesn’t do that anymore on this machine, it just dies silently. This is on a Mac running 10.11.6 and the DesktopServer/XAMPP PHP binary. I don’t actually do dev stuff daily so I can’t be sure of what’s changed since WP-CLI was correctly outputting error messages. I also don’t have this problem when I ssh into my host and use 0.24.1 there.
